Document next_start calculation on job failure (#2329)
timescale/docs: api/alter_job.md

```sql
SELECT alter_job(1000, next_start => '2020-03-15 09:00:00.0+00');
```

### Calculation of next start on failure

<Highlight type="note">
When a job run results in a runtime failure, the next start of the job is calculated taking into account both its `retry_period` and `schedule_interval`.

> The next_start is calculated using the following formula:

Author: updated as requested

```
next_start = finish_time + consecutive_failures * retry_period
```
plus some jitter (± 13%) to avoid the “thundering herds” effect.

> However, as we don’t want to put off the next start indefinitely or produce timestamps so large they end up out of range, we cap this at 5* schedule_interval, and we also do not consider more than 20 consecutive failures, so if the number of consecutive failures is higher, we will multiply by 20 there.

Reviewer: ```suggestion
However, to ensure that the `next start time` is not put off indefinitely or produce timestamps so large they end up out of range, it is capped at 5* `schedule_interval`. Also, more than 20 consecutive failures are not considered, so if the number of consecutive failures is higher, then it multiplies by 20.
```

Author: updated as requested

> Additionally in the case of jobs with fixed schedules, we make sure that if the next start calculated as above, surpasses the next scheduled execution, then the job is executed again at the next scheduled slot and not after that, in order to ensure that the job does not miss scheduled executions.

Reviewer: ```suggestion
Additionally, in the case of jobs with fixed schedules, the system ensures that if the next start calculated as specified, surpasses the next scheduled execution, then the job is executed again at the next scheduled slot and not after that. This ensures that the job does not miss scheduled executions.
```

> Finally, we distinguish between runtime failures that do not cause the job to crash and job crashes.

Reviewer: ```suggestion
Finally, there is a distinction between runtime failures that do not cause the job to crash and job crashes.
```

In the event of a job crash, the next start calculation still follows the above formula,

> but it will always be at least 5 minutes after the job's last finish, to give an operator enough time to disable it before another crash.

Reviewer: ```suggestion
but it is always be at least 5 minutes after the job's last finish, to give an operator enough time to disable it before another crash.
```

```sql
SELECT alter_job(1000, next_start => '2020-03-15 09:00:00.0+00');
```

### Calculation of next start on failure

<Highlight type="note">
When a job run results in a runtime failure, the next start of the job is calculated taking into account both its `retry_period` and `schedule_interval`.
The `next_start` time is calculated using the following formula:
```
next_start = finish_time + consecutive_failures * retry_period
```

> plus some jitter (± 13%) to avoid the “thundering herds” effect.

Author: It feels to me that this phrasing implies that adding jitter is up to the user, when in fact is always added and the user can't control it. So, I wanted to confirm with you before making the suggested change whether that is the meaning we convey here or not?

> plus some jitter (± 13%) to avoid the “thundering herds” effect.

Reviewer 2: Watch those smart quotes, too.

```suggestion
plus some jitter (± 13%) to avoid the "thundering herds" effect.
```

> plus some jitter (± 13%) to avoid the “thundering herds” effect.

Reviewer: > It feels to me that this phrasing implies that adding jitter is up to the user, when in fact is always added and the user can't control it. So, I wanted to confirm with you before making the suggested change whether that is the meaning we convey here or not?

In that case, maybe you want to add it as a part of the calculation formula.

> plus some jitter (± 13%) to avoid the “thundering herds” effect.

Author: I made a slight change there

> 

Reviewer: <Highlight type="note">

> 

Reviewer: <Highlight type="note">

> However, to ensure that the `next_start` time is not put off indefinitely or produce timestamps so large they end up out of range, it is capped at 5*`schedule_interval`.

Reviewer: ```suggestion
To ensure that the `next_start` time is not put off indefinitely or produce timestamps so large they end up out of range, it is capped at 5*`schedule_interval`.
```
